design-principles.md

```diff
@@ -8,14 +8,65 @@ permalink: /design-principles/
 * TOC Placeholder
 {:toc}
 
-## Introduction
+Introduction
+------------
 
 Flutter is written based on some core principles that were mostly
 intuited from past experiences with other platforms such as the Web
 and Android, some of which are summarised below.
 
 This document is most useful if you want to contribute to Flutter, as
-then we hope you will also follow these philosophies.
+then we hope you will also follow these philosophies. Please also read
+our
+[style guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo)
+for more specific guidelines regarding writing Dart code for Flutter.
+
+
+Code review and checking in code
+--------------------------------
+
+Every PR must be code-reviewed before check-in, including things like
+rolling a dependency. Getting a review means that a regular Flutter
+contributor (someone with commit access) has written a comment saying
+"LGTM" on your PR, and you have addressed all their feedback. ("LGTM"
+means "Looks Good To Me".) If you review a patch, you are sharing the
+responsibility for the patch with its author.
+
+Reviewers should not give an LGTM unless the patch has tests that
+verify all the affected code, or unless a test would make no sense.
+
+A reviewer may in some circumstances consider the code satisfactory
+without having fully reviewed or understood it. If a reviewer has not
+fully reviewed the code, they must admit to this by saying "RSLGTM"
+rather than just "LGTM". If you feel your code needs a real review,
+please find someone to actually review it. ("RSLGTM" means "Rubber
+Stamp Looks Good To Me".) If you mark a patch as RSLGTM, you are still
+sharing the responsibility for the patch with its author.
+
+If you really need to check something in in a hurry, for example
+because everything is broken and you can fix it, then pick someone on
+the team who you want to have review the code, and then mark the PR as
+"TBR" with their name. ("TBR" means "To Be Reviewed".) This is only to
+be used in emergencies. Nobody being around to review your 50,000 line
+patch at midnight on December 31st is not an emergency.
+
+Run the tests before checking code in. Travis does this for you, so
+wait for Travis to give the green light before merging a PR. The only
+time it is ok to skip this step is if the tests are already failing,
+and your fix is intended to improve matters, and you're going to watch
+the results once you've checked it in (in which case the bots are
+essentially doing the same job as Travis would have on your PR).
+
+Make sure all the trees and dashboards are green before checking in:
```

Comment:
Really?  I'm supposed to check 5 sites before committing?  I'm glad we're writing it down, but seems like we should fix this.